src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/OnTypeFormattingHandler.cs

```diff
@@ -113,6 +120,29 @@ public async Task<TextEdit[]> HandleRequestAsync(DocumentOnTypeFormattingParams
             return remappedEdits;
         }
 
+        private async Task<RazorLanguageKind?> GetTriggerCharacterLanguageKindAsync(LSPDocumentSnapshot documentSnapshot, Position positionAfterTriggerChar, string triggerCharacter, CancellationToken cancellationToken)
+        {
+            // request.Character will point to the position after the character that was inserted.
+            // For onTypeFormatting, it makes more sense to look up the projection of the character that was inserted.
```

Comment:
Looks like you don't do this anymore correct?

Reply:
I still do. But I no longer do `CharacterPosition - 1`. Instead, I find the correct position in a more complicated way.